lib/faye-redis-sharded.js

```diff
@@ -289,20 +289,34 @@ Engine.prototype = {
         return self._ns + '/channels' + c;
       });
 
-    var notify = function (error, clients) {
-      if (error) return;
+    var notify = function (error, clientIds) {
+      if (error) {
+        return;
+      }
 
-      clients.forEach(function (clientId) {
-        var shard = self._getShard(clientId, shardManager),
-          redis = shard.redis;
+      const processNotification = (clientId) => {
+        const shard = self._getShard(clientId, shardManager);
+        const redis = shard.redis;
         self._server.debug('Queueing for client ?: ?', clientId, message);
-        redis.rpush(self._ns + '/clients/' + clientId + '/messages', jsonMessage, function (err, result) {
-          redis.publish(self._ns + '/' + clientId + '/notify', clientId);
+        redis.rpush(`${self._ns}/clients/${clientId}/messages}`, jsonMessage, function (err, result) {
+          redis.publish(`${self._ns}/${clientId}/notify`, clientId);
           self._server.debug('Published for client ? - ? - to server ?', clientId, message, shard.shardName);
         });
         self.clientExists(clientId, function(exists) {
-          if (!exists) redis.del(self._ns + '/clients/' + clientId + '/messages');
+          if (!exists) {
+            redis.del(`${self._ns}/clients/${clientId}/messages`);
+          }
+        });
+      };
+
+      const start = new Date();
+      clientIds.reduce((promiseChain, clientId) => {
+        return promiseChain.then(() => {
+          processNotification(clientId);
         });
+      }, Promise.resolve()).finally(() => {
+        const end = new Date();
+        console.log(`[notify] Notified ${clientIds.length} clients in ${end - start}ms`);
```

Comment:
Should we write this to fluentd instead of (or in addition to) CloudWatch logs? Or will we drop the logging altogether when you create a CloudWatch metric for this?

Reply:
Planning to use these log messages in CloudWatch to create a metric. I'd prefer to remove these after we get additional information out of production.